saleor/payment/models.py

```diff
@@ -151,19 +157,25 @@ def can_authorize(self):
             self.is_active and self.charge_status == ChargeStatus.NOT_CHARGED)
 
     def can_capture(self):
-        # FIXME should also have an auth transaction
         not_charged = self.charge_status == ChargeStatus.NOT_CHARGED
+        is_authorized = self.transactions.filter(
```

Comment:
Have you tested how these changes affect database performance? Each call to `transactions.filter()` or `exists()` will perform a database query (this applies to other changed lines in this PR as well). This may result in duplicated queries and slow down our views or API calls.

We could prevent that if we ensured that the transactions are prefetched at some point earlier and iterate over them here to take advantage of prefetch cache:
```
is_authorized = any([txn.kind == TransactionKind.AUTH and txn.is_success for txn in self.transactions.all()])
```
Something like that should do the trick.

Reply:
Very good suggestions, I will think about where to pretech the transactions and how to update the cache when new transaction is added.

Reply:
Usually, we do the prefetching in the views as this is most often the "entry point" to a particular flow and it starts with fetching necessary data from DB. Every other function/method that performs some logic could safely assume that data was already prefetched at this point.

Reply:
I've updated the PR. The db query is optimized by cached transactions, and the cached transactions will be invalidated when new transactions are added or deleted. Tests are also added. I've tested locally and the db queries looks fine. How about this approach?